File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.ts
function minAddToMakeValid(s: string): number {
    let stack: string[] = [];
    for (const c of s){
        if (stack.length > 0 && stack[stack.length - 1] === '(' && c === ')'){
            stack.pop();
        } else {
            stack.push(c);
        }
    }
    return stack.length
};

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        stack = []
        val = 0
        for c in s:
            if c == "(":
                stack.append(c)
            elif stack:
                stack.pop()
            else:
                val += 1
        return len(stack) + val
            
        